units: Add standard_check fuzz test

mpbagot opened the pull request:
Introduce fuzzing using the `standard_test` crate for all structs in units, sans Error types.

Co-developed-by: Tobin C. Harding \<me@tobin.cc>

Close: #4252

tcharding commented:
Everything looks good except we cannot add `Default` impls, these were left out on purpose because these types don't have a meaningful default. We can feature gate an impl though, e.g:

```rust
    #[cfg(fuzzing)]
    impl Default for BlockTime {
        fn default() -> Self { Self(u32::default()) }
    }
```

Which is equivalent to

```rust
    #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
    #[cfg_attr(fuzzing, derive(Default))]
    pub struct BlockTime(u32);
```

nyonson commented:
> I'm confused, why did you mention `kani`?

Yea, sorry, I went down a little bit of a rabbit hole of what counts as "fuzzing" since this feels more like validation than my naive understanding of fuzzing (toss bytes at a function to see if it explodes). That rabbit hole lead me to these descriptions of other tools compared to kani which were helpful for my mind to see the responsibility boundaries.

tcharding commented:
I _think_ this is correct because I don't know of a better way to do it. But it feels a bit gross to have users pull in an additional dependency when they build with `--all-features` for something that seems like it should be totally internal to our project. @apoelstra can you think of a better way to do this?

tcharding reviewed (dismissed):
ACK cdca6ed95a9c09c81d5331555aecdd4a2342a9fe

mpbagot commented:
I missed the instructions in https://github.com/rust-bitcoin/rust-bitcoin/tree/master/fuzz#adding-fuzz-tests
I've now made the adjustments to the generate-files.sh that were needed, updated the generated files and rebased on top of master while I was at it.
